### ISSUE TITLE
Failed to inject profiler into PID

### ISSUE THREAD

**Issue by KL-Ding** [jvm-profiling-tools/async-profiler#620]
I have profiler and java application container running in a pod.
And the `ps -aux `in the pod is as follows:

USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
65535          1  0.0  0.0    972     4 ?        Ss   Jul20   0:00 /pause
root           7  0.0  1.5 2963580 119216 ?      Ssl  Jul20   0:33 java -jar /app/perf-demo.jar
root          22  0.0  0.0   2420    68 ?        Ss   Jul20   0:00 /bin/sh -c sleep inf
root          28  0.0  0.0   2332    60 ?        S    Jul20   0:00 sleep inf
root          44  0.0  0.0   4100  1924 pts/0    Ss+  Jul20   0:00 bash
root          58  0.0  0.0   4100  3388 pts/1    Ss   02:24   0:00 bash
root          84  0.0  0.0   6700  2892 pts/1    R+   02:25   0:00 ps -aux

However, when I try to run profiler to profile with PID 7, I got "Failed to inject profiler into 7".

**Comment by apangin** [comment]
Do you run profiler inside the same container as JVM runs?
Profiler library must be accessible by the target process.

This issue is mentioned in the [Troubleshooting](https://github.com/jvm-profiling-tools/async-profiler#troubleshooting) section.

**Comment by KL-Ding** [comment]
It is to run the JVM and the profiler in two containers under a Pod.
The container the profiler runs on is an ephemeral container.

**Comment by apangin** [comment]
I'm closing this as a duplicate of #78, #601, #166, #203.
Let me know if the linked issues still don't help.

**TL;DR** make sure the profiler library `libasyncProfiler.so` is accessible inside the target container. If async-profiler is on the different path inside the container, use `--lib` option.
